cpp/models/sde_seirvv/parameters.h

```diff
@@ -0,0 +1,303 @@
+struct TimeExposedV2 {
+    using Type = UncertainValue<ScalarType>;
+    static Type get_default()
+    {
+        return Type(6.0);
+    }
+    static std::string name()
+    {
+        return "TimeExposedV2";
+    }
+};
+
+/**
+ * @brief The infectious time of variant 1 in day unit.
+ */
+struct TimeInfectedV1 {
+    using Type = UncertainValue<ScalarType>;
+    static Type get_default()
+    {
+        return Type(6.0);
+    }
+    static std::string name()
+    {
+        return "TimeInfectedV1";
+    }
+};
+
+/**
+ * @brief The infectious time of variant 2 in day unit.
+ */
+struct TimeInfectedV2 {
+    using Type = UncertainValue<ScalarType>;
+    static Type get_default()
+    {
+        return Type(6.0);
+    }
+    static std::string name()
+    {
+        return "TimeInfectedV2";
+    }
+};
+
+
+
+/**
+ * @brief The contact patterns within the society are modelled using a ContactMatrix.
+ */
+struct ContactPatterns {
+    using Type = ContactMatrix;
+    static Type get_default()
+    {
+        return Type{1};
+    }
+    static std::string name()
+    {
+        return "ContactPatterns";
+    }
+};
+
+using ParametersBase = ParameterSet<TransmissionProbabilityOnContactV1, TransmissionProbabilityOnContactV2, 
+    TimeExposedV1, TimeExposedV2, TimeInfectedV1, TimeInfectedV2, ContactPatterns>;
+
+/**
+ * @brief Parameters of stochastic SEIRVV model.
+ */
+class Parameters : public ParametersBase
+{
+public:
+    Parameters()
+        : ParametersBase()
+    {
+    }
+
+    /**
+     * @brief Checks whether all Parameters satisfy their corresponding constraints and applies them, if they do not.
+     * Time spans cannot be negative and probabilities can only take values between [0,1]. 
+     *
+     * Attention: This function should be used with care. It is necessary for some test problems to run through quickly,
+     *            but in a manual execution of an example, check_constraints() may be preferred. Note that the apply_constraints()
+     *            function can and will not set Parameters to meaningful values in an epidemiological or virological context,
+     *            as all models are designed to be transferable to multiple diseases. Consequently, only acceptable
+     *            (like 0 or 1 for probabilities or small positive values for time spans) values are set here and a manual adaptation
+     *            may often be necessary to have set meaningful values.
+     *
+     * @return Returns true if one ore more constraint were corrected, false otherwise.  
+     */
+    bool apply_constraints()
+    {
+        double tol_times = 1e-1;
+
+        int corrected = false;
+        if (this->get<TimeExposedV1>() < tol_times) {
+            log_warning("Constraint check: Parameter TimeInfected changed from {:.4f} to {:.4f}. Please note that "
+                        "unreasonably small compartment stays lead to massively increased run time. Consider to cancel "
+                        "and reset parameters.",
```

Comment:
That is a lot of TimeInfected in these log messages - please update the names, and verify that each message as well as the enforced bounds actually apply to the respective parameter.

Reply:
This applies to check_constraints as well.